exp/ipam/api/v1alpha1/conversion.go

```diff
@@ -0,0 +1,94 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	"sigs.k8s.io/controller-runtime/pkg/conversion"
+
+	ipamv1beta1 "sigs.k8s.io/cluster-api/exp/ipam/api/v1beta1"
+	utilconversion "sigs.k8s.io/cluster-api/util/conversion"
+)
+
+func (src *IPAddress) ConvertTo(dstRaw conversion.Hub) error {
+	dst := dstRaw.(*ipamv1beta1.IPAddress)
+
+	if err := Convert_v1alpha1_IPAddress_To_v1beta1_IPAddress(src, dst, nil); err != nil {
+		return err
+	}
+
+	// unmarshalling of DataAnnotation is not done as v1beta1 and v1alpha1 are identical.
+	delete(src.Annotations, utilconversion.DataAnnotation)
+
+	return nil
+}
+
+func (dst *IPAddress) ConvertFrom(srcRaw conversion.Hub) error {
+	src := srcRaw.(*ipamv1beta1.IPAddress)
+
+	if err := Convert_v1beta1_IPAddress_To_v1alpha1_IPAddress(src, dst, nil); err != nil {
+		return err
+	}
+
+	return utilconversion.MarshalData(src, dst)
```

Comment:
Do we really need to call MarshalData here if we aren't using it?

Reply:
no, we do not.  i just removed it.